pkg/cmd/root/help_topic_test.go

```diff
@@ -4,76 +4,83 @@ import (
 	"testing"
 
 	"github.com/cli/cli/v2/pkg/iostreams"
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestNewHelpTopic(t *testing.T) {
+func findHelpTopic(t *testing.T, name string) helpTopic {
+	t.Helper()
+
+	for _, ht := range HelpTopics {
+		if ht.name == name {
+			return ht
+		}
+	}
+
+	require.FailNowf(t, "", "expected to find a topic with name: %q in topics: %v", name, HelpTopics)
+	return helpTopic{}
+}
+
+func TestEnvironmentHelp(t *testing.T) {
+	t.Parallel()
+
 	tests := []struct {
-		name     string
-		topic    string
-		args     []string
-		flags    []string
-		wantsErr bool
+		name      string
+		topic     string
+		args      []string
+		flags     []string
+		assertion require.ErrorAssertionFunc
```

Comment:
Something for discussion, providing the `assertion` here directly instead of a `bool` pushes the conditional out of the test. I've seen a lot of `assert.XYZ` without `return` in my time. 

At the least, I don't think there's ever a good time to use `assert.Error` or `assert.NoError` instead of `require`.

Reply:
This makes sense to me in this use case. In other use cases would you create a slice of assertions if there were multiple assertions that need to be made?

Reply:
No, probably not. I think the complexity in the setup would outweigh the benefit in terms of readability. Testify suggests it can be used for comparisons, errors and panics e.g. the example for https://pkg.go.dev/github.com/stretchr/testify/assert#ComparisonAssertionFunc

Like IIFEs, I feel 2/10 strongly about it. I think it's a neat pattern and low risk of causing issues though. Easy to back out if we didn't like the results of experimenting with it.